src/esp/io/JsonEspTypes.cpp

```diff
@@ -49,6 +49,16 @@ JsonGenericValue toJsonValue(const gfx::replay::Keyframe& keyframe,
     io::addMember(obj, "userTransforms", userTransformsArray, allocator);
   }
 
+  if (keyframe.lights && !keyframe.lights->empty()) {
+    JsonGenericValue lightsArray(rapidjson::kArrayType);
```

Comment:
We shouldn't have to write all this tedious serialization. `io::addMember` should already "just work" for Corrade::Optional and std::vector. Can this be simplified to `io::addMember(obj, keyframe.lights)`?

Reply:
I changed the approach to include a dirty flag, `lightsChanged`, instead of using `Corrade::Optional` to wrap the vector.